src/app/public/modules/modal/modal-scroll-shadow.directive.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  MutationObserverService
+} from '@skyux/core';
+
+import {
+  SkyTheme,
+  SkyThemeService
+} from '@skyux/theme';
+
+import {
+  SkyModalScrollShadowEventArgs
+} from './modal-scroll-shadow-event-args';
+
+/**
+ * @internal
+ */
+@Directive({
+  selector: '[skyModalScrollShadow]'
+})
+export class SkyModalScrollShadowDirective implements OnInit, OnDestroy {
+
+  @Output()
+  public skyModalScrollShadow = new EventEmitter<SkyModalScrollShadowEventArgs>();
+
+  private mutationObserver: MutationObserver;
+
+  private currentTheme: SkyTheme;
```

Comment:
Alphabetical ordering of properties is off here.

Reply:
Fixed.